ipatests/azure/azure-pipelines.yml

```diff
@@ -79,13 +79,13 @@ jobs:
         set -e
         echo "Running make target 'lint'"
         make V=0 lint
-        echo "Building documentation"
-        make -C doc/ html
-      displayName: Lint sources and documentation
-    - template: templates/save-test-artifacts.yml
+      displayName: Lint sources
+    - template: templates/${{ variables.BUILD_DOCS_TEMPLATE }}
+    - template: templates/publish-build.yml
       parameters:
-        logsPath: doc/_build/html
-        logsArtifact: docs-$(System.JobIdentifier)-$(Build.BuildId)-$(System.StageAttempt)-$(System.PhaseAttempt)-$(System.JobPositionInPhase)
+        artifactName: 'docs-$(Build.BuildId)-$(Agent.OS)-$(Agent.OSArchitecture)'
```

Comment:
In the logsArtifact previously we accounted for CI re-runs in the name. Would this change break when a docs artifact from the previous runs would already exist if you'd a re-run?
With this change we have tree artifacts that cannot be produced by individual re-runs:
![Screenshot from 2021-03-24 11-39-51](https://user-images.githubusercontent.com/2066/112288103-b0410900-8c95-11eb-8859-cd20dce7a7bf.png)

Reply:
It takes some time to investigate, sorry.
Previosly, the docs were published with the help of `templates/save-test-artifacts.yml` template, which unconditionally attempts to upload artifact. This may fail on Job re-run on the duplication of artifact name.
Current `templates/publish-build.yml` uploads files on succeeded dependency. The docs uploading step is the last step in job. So, currently, re-run of Lint task cannot fail due to docs name duplication. But as you mentioned this can actually happen for packages uploading. For preventing such scenario in future the unique artifact name should be used everywhere. I trigger re-run to prove the correctness of  produces artifact names.

Reply:
Ok, good.

Reply:
re-run attempts: https://dev.azure.com/freeipa/freeipa/_build/results?buildId=8323&view=results